src/components/bottom-bar/bottom-bar.tsx

```diff
@@ -99,6 +99,7 @@ class BottomBar extends Component<any, any> {
     const currentControlsWidth = this._getControlsWidth(resizeObserverEntry.target as HTMLElement);
     this._maxControlsWidth = Math.max(this._maxControlsWidth, currentControlsWidth);
     if (barWidth !== this.currentBarWidth) {
+      document.dispatchEvent(new Event('bottomBarClientRectChanged'));
```

Comment:
looks good !
lets make it a bit more organized
- make it a FakeEvent like GuiResizeEvent
- add it to EventType
- fire it on the player and not the document